[4.0] Fix ambiguity between overlay inits and default inits for BNNS structs.

stephentyrone opened the pull request:
Most of these are resolvable by simply switching over to inits without the previously-defaulted arguments, because the default args are all-or-nothing. In one case that doesn't make sense, so instead I've provided a dedicated static func to construct one class of object, and an init that doesn't match the importer-created one for the more common case.

resolves rdar://problem/33880553

[for swift-4.0-branch]
**Explanation:** Some inits in the BNNS overlay are not available due to ambiguity with the importer-created inits.
**Scope:** Local to the Accelerate/vecLib/BNNS module.
**Radar:** 33880553
**Risk:** Low. Only effects BNNS functions, and only really makes functionality available that is currently entirely broken.

stephentyrone committed:
Fix ambiguity between overlay inits and default inits for BNNS structs.

Most of these are resolvable by simply switching over to inits without
the previously-defaulted arguments, because the default args are all-
or-nothing. In one case that doesn't make sense, so instead I've provided
a dedicated static func to construct one class of object, and an init that
doesn't match the importer-created one for the more common case.

resolves rdar://problem/33880553

moiseev commented:
@swift-ci Please test

moiseev commented on test/stdlib/Accelerate.swift:
```diff
@@ -1,22 +1,27 @@
 // RUN: %target-run-simple-swift
 // REQUIRES: executable_test
 
-// UNSUPPORTED: OS=watchos
```
This one caused [problems in master](https://github.com/apple/swift/pull/11580).

stephentyrone committed:
Mark watchOS unsupported in Accelerate tests.

These APIs are actually available, but this breaks in automated
testing for reasons that we haven't had time to investigate; revert
to marking them unsupported for now.

stephentyrone commented:
@swift-ci Please test
